src/enclave/rpcendpoint.h

```diff
@@ -34,39 +35,93 @@ namespace enclave
       session_id(session_id)
     {}
 
+    auto split_actor_and_method(const std::string& actor_method)
+    {
+      const auto split_point = actor_method.find_last_of('/');
+      const auto actor = actor_method.substr(0, split_point);
+      const auto method =
+        actor_method.substr(split_point + 1, actor_method.size());
+
+      // NB: If the string does not contain '/', then both actor and method will
+      // contain the entire string
+      return std::make_pair(actor, method);
+    }
+
     bool handle_data(const std::vector<uint8_t>& data)
     {
-      if (!handler)
+      LOG_TRACE_FMT("Entered handle_data {} {}", data.size(), data.empty());
+
+      const SessionContext session(session_id, peer_cert());
+      RPCContext rpc_ctx(session);
+
+      rpc_ctx.raw = data;
+
+      // If we are unable to detect packing format, default to sending responses
+      // as Text
+      rpc_ctx.pack = jsonrpc::Pack::Text;
+
+      auto [success, rpc] = jsonrpc::unpack_rpc(data, rpc_ctx.pack);
+
+      if (!success)
+      {
+        send(jsonrpc::pack(rpc, rpc_ctx.pack.value()));
+        return true;
+      }
+      LOG_TRACE_FMT("Deserialised");
+
+      parse_rpc_context(rpc_ctx, rpc);
+
+      auto prefixed_method = rpc_ctx.method;
+      if (prefixed_method.empty())
       {
-        // The hostname indicates the rpc class.
-        auto host = hostname();
+        send(jsonrpc::pack(
+          jsonrpc::error_response(
+            rpc_ctx.seq_no,
+            jsonrpc::StandardErrorCodes::METHOD_NOT_FOUND,
+            "No method specified"),
+          rpc_ctx.pack.value()));
+        return true;
+      }
+      LOG_TRACE_FMT("Got method: {}", prefixed_method);
 
-        actor = rpc_map->resolve(host);
-        if (actor == ccf::ActorsType::unknown)
-          return false;
+      // Separate JSON-RPC method into actor and true method
+      auto [actor_s, method] = split_actor_and_method(prefixed_method);
+      rpc_ctx.method = method;
 
-        auto search = rpc_map->find(actor);
-        if (!search.has_value())
-          return false;
+      auto actor = rpc_map->resolve(actor_s);
+      if (actor == ccf::ActorsType::unknown)
+      {
+        send(jsonrpc::pack(
+          jsonrpc::error_response(
+            rpc_ctx.seq_no,
+            jsonrpc::StandardErrorCodes::METHOD_NOT_FOUND,
+            fmt::format("No such prefix: {}", actor_s)),
+          rpc_ctx.pack.value()));
+        return true;
+      }
+      rpc_ctx.actor = actor;
 
-        // If there is a client cert, pass it to the rpc handler.
-        LOG_DEBUG_FMT("RPC endpoint {}: {}", session_id, host);
-        handler = search.value();
-        caller = peer_cert();
+      auto search = rpc_map->find(actor);
+      if (!search.has_value())
+      {
```

Comment:
I think we should at least LOG_DEBUG here?

Reply:
Sure - I've reinstated a `LOG_DEBUG` similar to the one we had previously.